source/geometry_model/interface.cc

```diff
@@ -65,6 +66,15 @@ namespace aspect
 
 
     template <int dim>
+    bool
+    Interface<dim>::point_is_in_domain(const Point<dim> &p) const
+    {
+      // return false in the base class
+      return false;
```

Comment:
That seems like the wrong approach. I would either throw an exception here, or just make the function abstract in the base class.